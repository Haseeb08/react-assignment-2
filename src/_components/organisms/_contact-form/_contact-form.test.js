import React from "react";
import { create } from "react-test-renderer";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ContactForm from "./_contact-form";

describe("About component testing", () => {
  test("Matches the snapshot", () => {
    const input = create(<ContactForm />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("error msgs for empty inputs", () => {
    const { getByRole ,getByTestId} = render(<ContactForm />);

    fireEvent.click(getByRole("button"));
    expect(getByTestId("emailError")).toBeInTheDocument();
  });

  test("error msgs for inavlid inputs", () => {
    const { getByPlaceholderText, getByRole ,getByTestId} = render(<ContactForm />);
    fireEvent.change(getByPlaceholderText("Enter name"), {
        target: { value: "123" },
      });
      fireEvent.change(getByPlaceholderText("email"), {
        target: { value: "ab" },
      });
      fireEvent.change(getByPlaceholderText("phone number"), {
        target: { value: "1234" },
      });
    fireEvent.click(getByRole("button"));
    expect(getByTestId("emailError")).toBeInTheDocument(); 
 });

 test("valid inputs", () => {
    const { getByPlaceholderText, getByRole ,getByTestId} = render(<ContactForm />);
    fireEvent.change(getByPlaceholderText("Enter name"), {
        target: { value: "Abc" },
      });
      fireEvent.change(getByPlaceholderText("email"), {
        target: { value: "ab@gmail.com" },
      });
      fireEvent.change(getByPlaceholderText("phone number"), {
        target: { value: "1234567890" },
      });
      fireEvent.change(getByPlaceholderText("Subject"), {
        target: { value: "abcd" },
      });
    fireEvent.click(getByRole("button"));
    expect(getByTestId("emailError").value).toBeUndefined(); 
 });

});
