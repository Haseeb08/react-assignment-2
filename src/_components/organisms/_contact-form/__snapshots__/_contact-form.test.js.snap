// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component testing Matches the snapshot 1`] = `
<div
  className="makeStyles-form-1"
>
  <form>
    <div
      className="makeStyles-firstName-2"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Name
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-7 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter name"
            type="text"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
        data-testid="emailError"
      />
    </div>
    <div
      className="makeStyles-firstName-2"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Email
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-7 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="email"
            type="email"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-firstName-2"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Phone number
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-7 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="phone number"
            type="text"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-firstName-2"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Subject
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-7 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Subject"
            type="text"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-button-4"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button1-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </form>
</div>
`;
